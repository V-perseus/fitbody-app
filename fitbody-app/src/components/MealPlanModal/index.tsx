import MealPlanModal from './MealPlanModal'

export default MealPlanModal
