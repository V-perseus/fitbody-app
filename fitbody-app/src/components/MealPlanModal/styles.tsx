import { StyleSheet } from 'react-native'
import globals from '../../config/globals'

const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalContents: {
    marginHorizontal: 32,
    borderRadius: 8,
    backgroundColor: globals.styles.colors.colorWhite,
    minHeight: 197,
    paddingTop: 39,
    paddingBottom: 25,
    paddingHorizontal: 25,
  },
  modalTitle: {
    fontFamily: globals.fonts.primary.boldStyle.fontFamily,
    fontSize: 17,
    textAlign: 'center',
    letterSpacing: -0.36,
    color: 'black',
  },
  modalButtons: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: 32,
  },
  modalButton1: {
    flex: 1,
    height: 55,
    borderWidth: 1,
    borderRadius: 8,
    borderColor: globals.styles.colors.colorBlack,
    backgroundColor: 'transparent',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalButton2: {
    flex: 1,
    height: 55,
    marginLeft: 12.5,
    borderRadius: 8,
    backgroundColor: globals.styles.colors.colorPink,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalButtonText: {
    fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
    fontSize: 16,
    textAlign: 'center',
    lineHeight: 22,
    letterSpacing: -0.39,
    color: globals.styles.colors.colorBlack,
  },
  textColor1: {
    color: globals.styles.colors.colorBlack,
  },
  textColor2: {
    color: globals.styles.colors.colorWhite,
  },
})

export default styles
