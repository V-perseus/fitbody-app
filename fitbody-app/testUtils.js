import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
