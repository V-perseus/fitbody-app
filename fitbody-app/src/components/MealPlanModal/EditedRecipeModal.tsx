import React from 'react'
import { View, Text } from 'react-native'
import Modal from 'react-native-modal'

// Assets
import EditedIcon from '../../../assets/images/svg/icon/16px/edited.svg'
import styles from './styles'
import globals from '../../config/globals'
import { ButtonSquare } from '../Buttons/ButtonSquare'
import { ButtonIcon } from '../Buttons/ButtonIcon'

interface IEditRecipeModalProps {
  showModal: boolean
  onClose: () => void
  noButtonPressHandler: () => void
  yesButtonPressHandler: () => void
}
const EditedRecipeModal: React.FC<IEditRecipeModalProps> = ({ showModal, onClose, noButtonPressHandler, yesButtonPressHandler }) => {
  return (
    <Modal isVisible={showModal} backdropOpacity={0.5} onBackdropPress={onClose} useNativeDriver={true} testID="modal_bg">
      <View style={[styles.modalContents, { paddingTop: 32, paddingHorizontal: 20 }]}>
        <Text style={{ fontFamily: globals.fonts.primary.boldStyle.fontFamily, fontSize: 18, textAlign: 'center' }}>
          You previously{'\n'}edited this recipe
        </Text>
        <Text style={{ fontFamily: globals.fonts.primary.style.fontFamily, fontSize: 14, marginTop: 16, textAlign: 'center' }}>
          Which version would you like to use?
        </Text>
        <View style={[styles.modalButtons, { marginTop: 24 }]}>
          <ButtonIcon
            style={{ ...styles.modalButton2, ...{ height: 48, paddingHorizontal: 10, marginLeft: 0 } }}
            text={'KEEP\nEDITED'}
            textStyle={{ ...styles.modalButtonText, ...styles.textColor2 }}
            useOpacity={true}
            onPress={noButtonPressHandler}
            leftIcon={() => <EditedIcon color={globals.styles.colors.colorWhite} style={{ marginRight: 8 }} />}
          />
          <ButtonSquare
            style={{ ...styles.modalButton1, ...{ height: 48, paddingHorizontal: 10, marginLeft: 12.5 } }}
            text="USE ORIGINAL"
            textStyle={{ ...styles.modalButtonText, ...styles.textColor1 }}
            onPress={yesButtonPressHandler}
          />
        </View>
      </View>
    </Modal>
  )
}

export default EditedRecipeModal
