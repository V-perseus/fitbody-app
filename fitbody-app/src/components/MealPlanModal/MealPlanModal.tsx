import React from 'react'
import { View, Text } from 'react-native'
import Modal from 'react-native-modal'

// Components
import { ButtonSquare } from '../Buttons/ButtonSquare'

// Assets
import styles from './styles'

interface IMealPlanModalProps {
  onClose: () => void
  showModal: boolean
  modalText: string
  noButtonText: string
  noButtonPressHandler: () => void
  yesButtonText: string
  yesButtonPressHandler: () => void
}
const MealPlanModal: React.FC<IMealPlanModalProps> = ({
  onClose,
  showModal,
  modalText,
  noButtonText,
  noButtonPressHandler,
  yesButtonText,
  yesButtonPressHandler,
}) => (
  <Modal isVisible={showModal} backdropOpacity={0.5} onBackdropPress={onClose} useNativeDriver={true} testID="modal_bg">
    <View style={styles.modalContents}>
      <Text style={styles.modalTitle}>{modalText}</Text>
      <View style={styles.modalButtons}>
        <ButtonSquare style={styles.modalButton1} text={noButtonText} textStyle={styles.modalButtonText} onPress={noButtonPressHandler} />
        <ButtonSquare
          style={styles.modalButton2}
          text={yesButtonText}
          textStyle={{ ...styles.modalButtonText, ...styles.textColor2 }}
          onPress={yesButtonPressHandler}
        />
      </View>
    </View>
  </Modal>
)

export default MealPlanModal
