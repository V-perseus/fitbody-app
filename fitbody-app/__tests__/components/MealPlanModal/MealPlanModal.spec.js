import 'react-native'
import React from 'react'

import { render, fireEvent } from '../../../testUtils'
import MealPlanModal from '../../../src/components/MealPlanModal'

describe('MealPlanModal', () => {
  it('renders snapshot correctly', () => {
    const tree = render(
      <MealPlanModal
        showModal={true}
        modalText={'Are you sure you want see a title?'}
        noButtonText={'NO'}
        yesButtonText={'YES'}
        noButtonPressHandler={jest.fn}
        yesButtonPressHandler={jest.fn}
        onClose={jest.fn}
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders content correctly', async () => {
    const { findByText } = render(
      <MealPlanModal
        showModal={true}
        modalText={'Are you sure you want see a title?'}
        noButtonText={'NO'}
        yesButtonText={'YES'}
        noButtonPressHandler={jest.fn}
        yesButtonPressHandler={jest.fn}
        onClose={jest.fn}
      />,
    )

    await findByText(/Are you sure you want see a title?/i)
    await findByText('NO')
    await findByText('YES')
  })

  it('handles button presses', async () => {
    const yes = jest.fn()
    const no = jest.fn()
    const { findByText, findByTestId } = render(
      <MealPlanModal
        showModal={true}
        modalText={'Are you sure you want see a title?'}
        noButtonText={'NO'}
        yesButtonText={'YES'}
        noButtonPressHandler={no}
        yesButtonPressHandler={yes}
        onClose={no}
      />,
    )

    const noBtn = await findByText('NO')
    const yesBtn = await findByText('YES')
    const modalBg = await findByTestId('modal_bg')
    fireEvent.press(noBtn)
    expect(no).toHaveBeenCalled()
    fireEvent.press(yesBtn)
    expect(yes).toHaveBeenCalled()
    fireEvent(modalBg, 'onBackdropPress')
    expect(no).toHaveBeenCalledTimes(2)
  })
})
