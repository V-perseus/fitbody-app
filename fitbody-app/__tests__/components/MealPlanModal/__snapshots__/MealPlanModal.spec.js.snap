// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealPlanModal renders snapshot correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="modal_bg"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "marginHorizontal": 32,
          "minHeight": 197,
          "paddingBottom": 25,
          "paddingHorizontal": 25,
          "paddingTop": 39,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 17,
            "letterSpacing": -0.36,
            "textAlign": "center",
          }
        }
      >
        Are you sure you want see a title?
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 32,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "opacity": 1,
              },
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "height": 56,
                "justifyContent": "center",
                "width": 114,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#333333",
                "borderRadius": 8,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                },
                Object {
                  "color": "#333333",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 16,
                  "letterSpacing": -0.39,
                  "lineHeight": 22,
                  "textAlign": "center",
                },
              ]
            }
          >
            NO
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "opacity": 1,
              },
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "height": 56,
                "justifyContent": "center",
                "width": 114,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#fa8dd5",
                "borderRadius": 8,
                "flex": 1,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "center",
                "marginLeft": 12.5,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 16,
                  "letterSpacing": -0.39,
                  "lineHeight": 22,
                  "textAlign": "center",
                },
              ]
            }
          >
            YES
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
